tests/shared/drive-constraints.spec.js

```diff
@@ -556,4 +556,229 @@ describe('Shared: DriveConstraints', function() {
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.OK,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.OK
+        });
+      });
+    });
+
+    describe('given the drive contains the image', () => {
+
+      it('should return the source error', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: `${this.mountpoint}${this.separator}rpi.img`,
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.CONTAINS_IMAGE
+        });
+      });
+    });
+
+    describe('given the drive is a system drive', () => {
+
+      it('should return the system drive error', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.SYSTEM
+        });
+      });
+    });
+
+    describe('given the drive is too small', () => {
+
+      it('should return the too small error', function() {
+        this.drive.size = 1000000;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.TOO_SMALL
+        });
+      });
+    });
+
+    describe('given the drive is locked', () => {
+
+      it('should return the locked drive error', function() {
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.LOCKED
+        });
+      });
+    });
+
+    describe('given the drive is smaller than the recommended size', () => {
+
+      it('should return the smaller than recommended size warning', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000,
+          recommendedDriveSize: 5000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.WARNING,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.NOT_RECOMMENDED
+        });
+      });
+    });
+
+    describe('given the image is null', () => {
```

Comment:
We should also check that it behaves as expected if e.g. the image is null, but the drive is locked